pinot-query-planner/src/main/java/org/apache/pinot/query/planner/physical/DispatchablePlanVisitor.java

```diff
@@ -37,10 +40,7 @@
 
 
 public class DispatchablePlanVisitor implements PlanNodeVisitor<Void, DispatchablePlanContext> {
-  public static final DispatchablePlanVisitor INSTANCE = new DispatchablePlanVisitor();
-
-  private DispatchablePlanVisitor() {
-  }
+  private final Set<MailboxSendNode> _visited = Collections.newSetFromMap(new IdentityHashMap<>());
```

Comment:
Given we have spools now, a mailbox send may be visited by this visitor twice. This keeps track of the mailbox send nodes that have been visited to do not visit them again. That is unnecessary and may be problematic (called code may assume each mailbox is visited only once. I don't remember the kind of errors this may produce, but probably I wouldn't have detected it in case it didn't fail :laughing: